eventuate-common-jdbc-micronaut-data/src/main/java/io/eventuate/common/jdbc/micronaut/data/EventuateJdbcMicronautDataFactory.java

```diff
@@ -16,7 +16,7 @@ public EventuateJdbcStatementExecutor eventuateCommonJdbcOperations(JdbcOperatio
   }
 
   @Singleton
-  public EventuateTransactionTemplate eventuateTransactionTemplate(TransactionalExecutor transactionalExecutor) {
-    return transactionalExecutor::executeInTransaction;
+  public EventuateTransactionTemplate eventuateTransactionTemplate(EventuateMicronautDataTransactionTemplate eventuateMicronautDataTransactionTemplate) {
+    return eventuateMicronautDataTransactionTemplate;
```

Comment:
This bean definition doesn't seem to do anything.

Reply:
EventuateTransactionTemplate and EventuateMicronautDataTransactionTemplate are different beans for micronaut

Reply:
Ok but if this were a Spring application this would be a very strange @Bean definition.